lib/tasks/audit_index.rb

```diff
@@ -0,0 +1,132 @@
+require 'csv'
+require 'pp'
+require 'io/console'
+
+module Tasks
+  class AuditIndex
+    def all_es_firm_ids
+      @all_es_firm_ids ||= find_all_es_firm_ids
+    end
+
+    def all_pg_firm_ids
+      @all_pg_firm_ids ||= find_all_pg_firm_ids
+    end
+
+    def in_es_not_pg
+      @in_es_not_pg ||= all_es_firm_ids - all_pg_firm_ids
```

Comment:
Sorry, like I said, this class is intended to be used from the console too. The memoization is for that use-case. Otherwise yes, we wouldn't need to bother.